internal/pkg/service/objectstorage/cloud/azure/container.go

```diff
@@ -0,0 +1,85 @@
+package azure
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+
+	"github.com/giantswarm/object-storage-operator/api/v1alpha1"
+)
+
+func (s AzureObjectStorageAdapter) existsContainer(ctx context.Context, bucket *v1alpha1.Bucket, storageAccountName string) (bool, error) {
```

Comment:
This is not a new function but I split the existing functions from storage.go into smaller files to be more manageable